src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -383,22 +386,49 @@ private void changeLabels() {
      * Adds a style class to the listview which changes its background to contain a loading spinning gif
      */
     @Override
-    protected void addPanelLoadingIndication() {
+    protected void setLoadingIndicator() {
         logger.info("Preparing to add panel loading indication");
         listView.getStyleClass().add("listview-loading");
-
-        // Remove the items in the issue list because they are not relevant to the filter anymore.
-        // This also makes the listview background visible, since we intend to show a loading indicator on the
-        // background.
-        listView.setItems(null);
     }
 
     /**
-     * Removes the style class that was added in addPanelLoadingIndicator() from the listview.
+     * Removes the style class that was added in setLoadingIndicator() from the listview.
      */
     @Override
-    protected void removePanelLoadingIndication() {
+    protected void removeLoadingIndicator() {
         logger.info("Preparing to remove panel loading indication");
         listView.getStyleClass().removeIf(cssClass -> cssClass.equals("listview-loading"));
     }
+
+    @Override
+    protected void showRepoOpeningIndicator(RepoOpeningEvent e) {
+        setTranslucentCellFactory();
+        setLoadingIndicator();
+    }
+
+    @Override
+    protected void hideRepoOpeningIndicator(RepoOpenedEvent e) {
+        removeLoadingIndicator();
+    }
+
```

Comment:
These two methods are very similar to the ones below, so better to merge them? We can probably use the methods' names below since they are more general.